projects/ngx-sub-form/src/lib/ngx-sub-form.component.ts

```diff
@@ -21,12 +21,17 @@ import {
   isNullOrUndefined,
   ControlsType,
   ArrayPropertyKey,
+  TypedAbstractControl,
+  TypedFormGroup,
 } from './ngx-sub-form-utils';
-import { FormGroupOptions, NgxFormWithArrayControls, OnFormUpdate, TypedFormGroup } from './ngx-sub-form.types';
+import { FormGroupOptions, NgxFormWithArrayControls, OnFormUpdate } from './ngx-sub-form.types';
 
-type MapControlFunction<FormInterface, MapValue> = (ctrl: AbstractControl, key: keyof FormInterface) => MapValue;
+type MapControlFunction<FormInterface, MapValue> = (
+  ctrl: TypedAbstractControl<any>,
+  key: keyof FormInterface,
+) => MapValue;
 type FilterControlFunction<FormInterface> = (
-  ctrl: AbstractControl,
+  ctrl: TypedAbstractControl<any>,
```

Comment:
same.

probably 

```suggestion
  ctrl: TypedAbstractControl<MapValue>,
```

?

Reply:
Same as above. not sure what this does and how to test it, if you say adding typing here is fine I'll make the necessary changes.

Reply:
After having a look at it I'd say that as the map / filter functions are built today its not possible to type them:
- The any needs to replaced with the type of the property of the FormInterface
- we do have the name of the property via `key` but its a runtime parameter and hance not available for typing
![image](https://user-images.githubusercontent.com/265338/75140451-3e436c80-56ef-11ea-9dc6-a84aad2163cf.png)

Also when looking where and how map / filter is used I'd say that typing them really doesn't add any benefit as the lib code does not make any decisions based on a specific type and the methods using those interfaces are not exposed.

We can try and refactor the Map/Filter code to be fully typed, but I'd opt for doing this in a separate pull request as this pull request really only changes typings which is fairly low risk vs changing actual implementation.